File: lib/java/opentoken/src/main/java/com/truveta/opentoken/attributes/validation/AttributeValidator.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.attributes.validation;

/**
 * A generic interface for attribute validation.
 */
public interface AttributeValidator {

    /**
     * Validates an attribute value.
     * 
     * @param value the attribute value.
     * 
     * @return <code>true</code> if the attribute is valid; <code>false</code>
     *         otherwise.
     */
    boolean eval(String value);
}

File: lib/python/opentoken/src/main/opentoken/attributes/validation/attribute_validator.py
"""
Copyright (c) Truveta. All rights reserved.
"""

from abc import ABC, abstractmethod


class AttributeValidator(ABC):
    """
    A generic interface for attribute validation.
    """

    @abstractmethod
    def eval(self, value: str) -> bool:
        """
        Validate an attribute value.

        Args:
            value: The attribute value

        Returns:
            True if the attribute is valid, False otherwise
        """
        pass
